access/handler.go

```diff
@@ -1179,16 +1255,32 @@ func (h *Handler) blockHeaderResponse(header *flow.Header, status flow.BlockStat
 }
 
 // buildMetadataResponse builds and returns the metadata response object.
-func (h *Handler) buildMetadataResponse() *entities.Metadata {
+// Expected error returns during normal operations:
+// - codes.Internal if an error is encountered getting the highest indexed height from indexer.
+func (h *Handler) buildMetadataResponse() (*entities.Metadata, error) {
 	lastFinalizedHeader := h.finalizedHeaderCache.Get()
 	blockId := lastFinalizedHeader.ID()
 	nodeId := h.me.NodeID()
 
-	return &entities.Metadata{
+	metadata := &entities.Metadata{
 		LatestFinalizedBlockId: blockId[:],
 		LatestFinalizedHeight:  lastFinalizedHeader.Height,
 		NodeId:                 nodeId[:],
 	}
+
+	if h.indexReporter != nil {
+		var err error
+		var highestIndexedHeight uint64
+		highestIndexedHeight, err = h.indexReporter.HighestIndexedHeight()
+		if err != nil &&
+			!errors.Is(err, indexer.ErrIndexNotInitialized) {
+			return nil, rpc.ConvertIndexError(err, lastFinalizedHeader.Height, "could not get highest indexed height")
+		}
+
+		metadata.HighestIndexedHeight = highestIndexedHeight
```

Comment:
we should treat the return value from `HighestIndexedHeight` as undefined when an error is returned, and explicitly set the value to 0. this makes the intention clear for the next reader, and avoids hard to debug issues in the future if the implementation ever changes.